athenz-identity-provider-service/src/test/java/com/yahoo/vespa/hosted/athenz/instanceproviderservice/AthenzInstanceProviderServiceTest.java

```diff
@@ -101,88 +89,42 @@
     public void provider_service_hosts_endpoint_secured_with_tls() throws Exception {
         String domain = "domain";
         String service = "service";
+
         AutoGeneratedKeyProvider keyProvider = new AutoGeneratedKeyProvider();
         PrivateKey privateKey = keyProvider.getPrivateKey(0);
         AthenzProviderServiceConfig config = getAthenzProviderConfig(domain, service, "vespa.dns.suffix");
-        ScheduledExecutorServiceMock executor = new ScheduledExecutorServiceMock();
+        SslContextFactory sslContextFactory =  AthenzInstanceProviderService.createSslContextFactory();
```

Comment:
Nice 👍  This unit test got a lot better after removing `ScheduledExecutorServiceMock`.

Reply:
Consider mocking the `SslContextFactory` instance instead of creating one from `AthenzInstanceProviderService.createSslContextFactory()`.

Reply:
This would be (at least for now) exact copy of the method in `AthenzInstanceProviderService`? I don't think it would be easier to create mockito mock to use this, and returning actual `SslContextFactory` would probably still need to set the same fields we do in real method...

Reply:
Okay, I agree to leave it if it's not easy to mock.